src/Tests/Microsoft.NET.Publish.Tests/RuntimeIdentifiersTests.cs

```diff
@@ -79,6 +83,55 @@ public void BuildWithRuntimeIdentifier()
             }
         }
 
+        //  Run on core MSBuild only as using a local packages folder hits long path issues on full MSBuild
+        [CoreMSBuildOnlyFact]
+        public void BuildWithUseCurrentRuntimeIdentifier()
+        {
+            var testProject = new TestProject()
+            {
+                Name = "BuildWithUseCurrentRuntimeIdentifier",
+                TargetFrameworks = "netcoreapp3.0",
+                IsSdkProject = true,
+                IsExe = true
+            };
+
+            var compatibleRid = EnvironmentInfo.GetCompatibleRid(testProject.TargetFrameworks);
+
+            testProject.AdditionalProperties["UseCurrentRuntimeIdentifier"] = "True";
+
+            //  Use a test-specific packages folder
+            testProject.AdditionalProperties["RestorePackagesPath"] = @"$(MSBuildProjectDirectory)\..\pkg";
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject);
+            var restoreCommand = new RestoreCommand(testAsset);
+
+            restoreCommand
+                .Execute()
+                .Should()
+                .Pass();
+
+            var buildCommand = new BuildCommand(testAsset);
+
+            buildCommand
+                .ExecuteWithoutRestore()
+                .Should()
+                .Pass();
```

Comment:
For this test I don't think there's any reason to run the restore command separately.  So just collapse it into a single `BuildCommand` using `Execute` instead of `ExecuteWithoutRestore`.